src/gui/properties/propertieswidget.cpp

```diff
@@ -517,65 +517,51 @@ void PropertiesWidget::loadUrlSeeds()
+    QString folderPath {nameIndex.data().toString()};
+    for (QModelIndex modelIdx = m_propListModel->parent(nameIndex); modelIdx.isValid(); modelIdx = modelIdx.parent())
+        folderPath.prepend(modelIdx.data().toString() + QChar('/'));
+
+    const QDir saveDir {m_torrent->savePath(true)};
+    const QString fullPath {Utils::Fs::expandPath(saveDir.absoluteFilePath(folderPath))};
+    return fullPath;
 }
 
-void PropertiesWidget::openFile(const QModelIndex &index) const
+void PropertiesWidget::openDoubleClickedItem(const QModelIndex &index) const
 {
-    int i = m_propListModel->getFileIndex(index);
-    const QDir saveDir(m_torrent->savePath(true));
-    const QString filename = m_torrent->filePath(i);
-    const QString filePath = Utils::Fs::expandPath(saveDir.absoluteFilePath(filename));
-    qDebug("Trying to open file at %s", qUtf8Printable(filePath));
-    // Flush data
-    m_torrent->flushCache();
-    Utils::Gui::openPath(filePath);
+    m_torrent->flushCache();  // Flush data
+    Utils::Gui::openPath(getFullPath(index));
 }
 
-void PropertiesWidget::openFolder(const QModelIndex &index, const bool containingFolder) const
+void PropertiesWidget::openItemContainingFolder(const QModelIndex &index) const
 {
-    QString absolutePath;
-    // FOLDER
-    if (m_propListModel->itemType(index) == TorrentContentModelItem::FolderType) {
-        // Generate relative path to selected folder
-        const QModelIndex nameIndex {index.sibling(index.row(), TorrentContentModelItem::COL_NAME)};
-        QStringList pathItems {nameIndex.data().toString()};
-        QModelIndex parent = m_propListModel->parent(nameIndex);
-        while (parent.isValid()) {
-            pathItems.prepend(parent.data().toString());
-            parent = m_propListModel->parent(parent);
-        }
-        if (pathItems.isEmpty())
-            return;
+    QString path;
 
-        const QDir saveDir(m_torrent->savePath(true));
-        const QString relativePath = pathItems.join('/');
-        absolutePath = Utils::Fs::expandPath(saveDir.absoluteFilePath(relativePath));
+    if (m_propListModel->itemType(index) == TorrentContentModelItem::FolderType) {
+        // we want to open the parent folder
+        const QModelIndex parentIndex {index.parent()};
+        path = parentIndex.isValid() ? getFullPath(parentIndex) : m_torrent->savePath(true);
     }
     else {
-        const int i = m_propListModel->getFileIndex(index);
-        const QDir saveDir(m_torrent->savePath(true));
-        const QString relativePath = m_torrent->filePath(i);
-        absolutePath = Utils::Fs::expandPath(saveDir.absoluteFilePath(relativePath));
+        path = getFullPath(index);
     }
 
-    // Flush data
-    m_torrent->flushCache();
+    m_torrent->flushCache();  // Flush data
 #ifdef Q_OS_MACOS
-    Q_UNUSED(containingFolder);
-    MacUtils::openFiles(QSet<QString> {absolutePath});
+    MacUtils::openFiles({path});
```

Comment:
Offtopic: Seems this function should be used internally by "opening" functions from `Utils::Gui`.

Reply:
I looked around in the codebase, it is a little complicated. I'll leave it for another time.